bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,249 @@
+                    case "rn":
+                        response.setRainSensor(Integer.parseInt(status) == 1 ? true : false);
+                        break;
+                    default:
+                        logger.debug("Unknown response: {}", command);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5, int zone6,
+            int zone7, int zone8) {
+        try {
+            if (sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5
+                    + ":" + zone6 + ":" + zone7 + ":" + zone8) == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandClear() {
+        try {
+            if (sendGet("/result.cgi?xr") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogin() throws IOException {
+        return sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password) != null;
+    }
+
+    public boolean commandLogout() {
+        try {
+            if (sendGet("/ergetcfg.cgi?m=o") == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return null;
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return null;
+        }
+
+        List<String> rVal = new BufferedReader(new StringReader(response.getContentAsString())).lines()
+                .collect(Collectors.toList());
+
+        return rVal;
+    }
+
+    private static EtherRainUdpResponse updBroadcast() {
+        DatagramSocket c;
+
+        // Find the server using UDP broadcast
+        try {
+            // Open a random port to send the package
+            c = new DatagramSocket();
```

Comment:
A `DatagramSocket` is something that should be carefully closed otherwise you might be wasting valueable resources of a system. Please use try-with-resource this will you in handling this correctly, see also: http://www.fortitudetec.com/blogs/2016/8/8/java-try-with-resources

Reply:
I think what you meant here was that I should use a finally block to close the datagram. I have made this change. Let me know if there is something further you would like.

Reply:
I agree with martinvw, you should be using a try-with-resources to handle the DatagramSocket. The link he provided shows an example of how to use try-with-resources on a DatagramSocket.

Reply:
No worries.......I took a closer look at what you were asking and I modified to use the template you mentioned.